files/usr/share/cinnamon/desklets/photoframe@cinnamon.org/desklet.js

```diff
@@ -90,6 +93,11 @@ class CinnamonPhotoFrameDesklet extends Desklet.Desklet {
 
         let info;
         while ((info = fileEnum.next_file(null)) != null) {
+            /* Skip hidden files / directories */
+            if (this.skip_hidden && info.get_name().indexOf(".") == 0) {
```

Comment:
Instead of checking for `.` use info.get_is_hidden and add `standard::is-hidden` to the enumerate_children() call above.

Another method of hiding files is to create a file `.hidden` in the folder, and adding filenames that you want hidden. `info.get_is_hidden` will account for that as well.